### PR TITLE
Standalone RPC implementation

### PR DESCRIPTION
**Pull request by sorpaas**
Based on #108. Work in progress. Do not merge.

### CODE REVIEW

**Comment by sorpaas** [comment]
@dulanov Have a primary look if you have time. Those four RPCs are implemented, while not yet tested.

File: src/lib.rs [hunk 1]
```diff
@@ -41,6 +42,12 @@ mod util;
 
 pub use self::core::*;
 pub use self::util::*;
+pub use self::rpc::start;
+
+/// Get the current Emerald version.
+pub fn version() -> &'static str {
+    "v0.0.1"
```
Comment (dulanov) [inline src/lib.rs]: From `main.rs`: `const VERSION: Option<&'static str> = option_env!("CARGO_PKG_VERSION");`

File: src/rpc/mod.rs [hunk 2]
```diff
@@ -105,9 +109,36 @@ fn inject_nonce(url: Arc<http::AsyncWrapper>, p: &Params, addr: &Address) -> Res
...
+            return futures::finished(value.unwrap()).boxed();
+        )
+    }
+
+    macro_rules! put_error {
+        ( $p:expr ) => (
+            return futures::failed($p).boxed();
+        )
+    }
+
```
Comment (dulanov) [inline src/rpc/mod.rs]: What do you think about extracting macros from the function upstairs?

**Comment by dulanov** [comment]
For me looks good, you are macro guru ;)

File: src/rpc/mod.rs (hunk 2 above)
Comment (sorpaas) [inline src/rpc/mod.rs]: Have put them in a separate `rpc/macros.rs` file. So looks clearer now. 😄

**Comment by sorpaas** [comment]
Right now the Account field for `emerald_newAccount` has not yet been optional. I'll fix that tomorrow and then merge this.

**Comment by sjmackenzie** [comment]
waiting on this to merge so I can split out `emerald-core` and `emerald-cli` crates.

**Comment by sorpaas** [comment]
@sjmackenzie You can merge when the tests are passed (i.e. you can click the merge button).

**Comment by sjmackenzie** [comment]
I cannot approve the PR as I'm not a reviewer.

**Comment by sjmackenzie** [comment]
oh, seems the "approved" status doesn't get reset once new commits come in. Let's see after a successful build.

**Pull request merged**